examples/c/ex_col_store.c

```diff
@@ -0,0 +1,555 @@
+            count++;
+
+            /* Increment the values of the rec_arr with the temp_arr values */
+            rec_arr[0] += temp;
+            rec_arr[1] += humidity;
+            rec_arr[2] += pressure;
+            rec_arr[3] += wind;
+            rec_arr[4] += feels_like_temp;
+        }
+    }
+
+    printf("Number of matching entries: %u \n", count);
+
+    /* Get the average values by dividing with the total number of records */
+    for (int i = 0; i < num_rec; i++) {
+        rec_arr[i] = rec_arr[i] / count;
+    }
+    /* List the average records */
+    printf("Average records for location RUS : \nTemp: %" PRIu8 ", Humidity: %" PRIu8
+           ", Pressure: %" PRIu16 ", Wind: %" PRIu8 ", Feels like: %" PRIu8 "\n",
+      rec_arr[0], rec_arr[1], rec_arr[2], rec_arr[3], rec_arr[4]);
+    scan_end_check(ret == WT_NOTFOUND);
+    error_check(loc_cursor->close(loc_cursor));
+}
+
+int
+main(int argc, char *argv[])
+{
+    WEATHER *w;
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+    WT_CURSOR *cursor;
+    WEATHER weather_data[N_DATA];
+    uint16_t start, end;
+    int min_temp_result, max_temp_result;
+
+    start = 1000;
+    end = 2000;
+    min_temp_result = 0;
+
+    /* Generating random data to populate the weather table */
+    generate_data(weather_data);
+
+    home = example_setup(argc, argv);
+
+    /* Establishing a connection */
+    error_check(wiredtiger_open(home, NULL, "create,statistics=(fast)", &conn));
+
+    /* Establishing a session */
+    error_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Create a table with columns and colgroup */
+    error_check(session->create(session, "table:weathertable",
+      "key_format=r,value_format=" WT_UNCHECKED_STRING(
+        5sHBBHBBHH5s) ",columns=(id,day,hour,temp,"
+                      "humidity,pressure,wind,"
+                      "feels_like_temp,loc_lat,loc_long,country),colgroups=(day_time,temperature,"
+                      "humidity_pressure,"
+                      "wind,feels_like_temp,location)"));
+
+    /* Create the colgroups */
+    error_check(session->create(session, "colgroup:weathertable:day_time", "columns=(day,hour)"));
+    error_check(session->create(session, "colgroup:weathertable:temperature", "columns=(temp)"));
+    error_check(session->create(
+      session, "colgroup:weathertable:humidity_pressure", "columns=(humidity,pressure)"));
+    error_check(session->create(session, "colgroup:weathertable:wind", "columns=(wind)"));
+    error_check(session->create(
+      session, "colgroup:weathertable:feels_like_temp", "columns=(feels_like_temp)"));
+    error_check(session->create(
+      session, "colgroup:weathertable:location", "columns=(loc_lat,loc_long,country)"));
+
+    /* Open a cursor on the table to insert the data ---[[[[ INSERT ]]]]--- */
+    error_check(session->open_cursor(session, "table:weathertable", NULL, "append", &cursor));
+    w = weather_data;
```

Comment:
I am not sure why we need an additional "w" pointer, and why we can't use `weather_data` directly? Am I missing something.